packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -93,58 +94,48 @@ export class Relayer {
           continue;
         }
         try {
-          await this._relayPendingDeposit(l1Token, relayableDeposit);
+          relayTransactions.push(await this._generateRelayTransactionForPendingDeposit(l1Token, relayableDeposit));
         } catch (error) {
-          this.logger.error({
-            at: "InsuredBridgeRelayer#Relayer",
-            message: "Unexpected error processing deposit",
-            error,
-          });
+          console.log("error", error);
+          this.logger.error({ at: "AcrossRelayer#Relayer", message: "Unexpected error processing deposit", error });
         }
       }
+      await this._processTransactionBatch(relayTransactions as any);
```

Comment:
Shouldn't we try-catch this in case `_processTransactionBatch` fails, and the relayer fails, but the disputer/finalizer would still succeed?

Reply:
these are sent separately to avoid this exact kind of flow. Originally when designing this I wanted to batch _all_ transactions to the `processTransactionBatch` method for one particular contract. I decided agains this as you'd need the exact kind of logic you describe. the PR in it's current form will only send one kind of transaction to one destination at a time in a batch.

Reply:
I don't think I fully understand, since `_processTransactionBatch` is outside a try-catch, won't this failure on line 104 cause the Finalizer/Disputer to be blocked?

Reply:
after discussing more we decided to add additional try catch blocks.